flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -165,6 +215,31 @@ <IN, OUT> void insertBlockingBuffer(StateRequest<K, IN, OUT> request) {
      * @param force whether to trigger requests in force.
      */
     void triggerIfNeeded(boolean force) {
-        // TODO: implement the trigger logic.
+        if (!force && stateRequestsBuffer.activeQueueSize() < batchSize) {
```

Comment:
Another thing comes to my mind. The timeout trigger is needed here. How about adding a TODO here?

Reply:
Thanks for the reminder👍, added.